[docs] Improve Upgrade plan docs (#6018)
mui/mui-x: docs/data/introduction/licensing/licensing.md

The features exclusive to the Premium version are marked with the <span class="plan-premium"></span> icon throughout the documentation.
:::

## Upgrading

> The npm packages of of any given plan is a **superset** of the npm packages of the plan below.

Reviewer: ```suggestion
The npm package of any given plan is a **superset** of the package on the plan below.
```

> So to upgrade, replace the [npm packages](#plans) you installed and imported with the ones of the plan above.

Reviewer: ```suggestion
So to upgrade, replace the [npm packages](#plans) and the components' imports with the ones from the target plan.
```

> For example with the Data Grid:

Reviewer: ```suggestion
For example, when you want to upgrade the Data Grid:
```

- **Upgrading from Community to Pro.**

  `@mui/x-data-grid-pro` is a superset of `@mui/x-data-grid`, so you can upgrade from the Community to the Pro plan like this:

  ```diff
  -import { DataGrid } from '@mui/x-data-grid';
  +import { DataGridPro } from '@mui/x-data-grid-pro';
  ```

  :::warning
  ⚠️ However, there is an exception to the superset rule. The default value of the `pagination` prop changes, [see the docs of the pagination](https://mui.com/x/react-data-grid/pagination/).
  :::

- **Upgrading from Pro to Premium.**

>   `@mui/x-data-grid-premium` is a superset of `@mui/x-data-grid-pro`, so you can upgrade from the Pro to the Pro Premium like this:.

Reviewer: ```suggestion
  `@mui/x-data-grid-premium` is a superset of `@mui/x-data-grid-pro`, so you can upgrade from Pro to Premium like this:
```

  ```diff
  -import { DataGridPro } from '@mui/x-data-grid-pro';
  +import { DataGridPremium } from '@mui/x-data-grid-premium';
  ```

  :::info
  If you are looking for upgrading from Pro to Premium, please contact us at [sales@mui.com](mailto:sales@mui.com?subject=My%20upgrade%20discount%20to%20Premium).

>   We'll provide you with a discount to recoup the remaining time of your Pro license.

Reviewer: ```suggestion
  We'll provide you with a discount based on the remaining time of your current license term.
```
